packages/material-ui/src/Badge/Badge.d.ts

```diff
@@ -6,7 +6,7 @@ export interface BadgeProps
   children: React.ReactNode;
   badgeContent?: React.ReactNode;
   color?: PropTypes.Color | 'error';
-  component?: React.ElementType<BadgeProps>;
+  component?: React.ElementType<React.HTMLAttributes<HTMLDivElement>>;
```

Comment:
See commit description of https://github.com/mui-org/material-ui/commit/1f2505aac1d5b59410ad9dfce57bc14f575d4513